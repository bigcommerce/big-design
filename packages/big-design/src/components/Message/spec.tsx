import { theme } from '@bigcommerce/big-design-theme';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import 'jest-styled-components';

import { Message, MessageProps } from './Message';

test('renders with margins', () => {
  const { container, rerender } = render(<Message messages={[{ text: 'Success' }]} />);

  expect(container.firstChild).not.toHaveStyle('margin: 1rem');

  rerender(<Message margin="medium" messages={[{ text: 'Success' }]} />);

  expect(container.firstChild).toHaveStyle('margin: 1rem');
});

test('render default (success) Message', () => {
  const { container } = render(<Message messages={[{ text: 'Success' }]} />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyle(
    `border-left: ${theme.spacing.xxSmall} solid ${theme.colors.success}`,
  );
});

test('render error Message', () => {
  const { container } = render(<Message messages={[{ text: 'Error' }]} type="error" />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyle(
    `border-left: ${theme.spacing.xxSmall} solid ${theme.colors.danger}`,
  );
});

test('render warning Message', () => {
  const { container } = render(<Message messages={[{ text: 'Warning' }]} type="warning" />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyle(
    `border-left: ${theme.spacing.xxSmall} solid ${theme.colors.warning50}`,
  );
});

test('render info Message', () => {
  const { container } = render(<Message messages={[{ text: 'Info' }]} type="info" />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyle(
    `border-left: ${theme.spacing.xxSmall} solid ${theme.colors.primary60}`,
  );
});

test('renders with link', async () => {
  const { container } = render(
    <Message messages={[{ text: 'Success', link: { text: 'Link', href: '#' } }]} />,
  );

  expect(container.firstChild).toMatchSnapshot();

  const link = await screen.findByRole<HTMLAnchorElement>('link');

  expect(link).toBeInTheDocument();
  expect(link.href).toBe('http://localhost/#');
});

test('renders with external link', async () => {
  const { container } = render(
    <Message
      messages={[
        { text: 'Success', link: { text: 'Link', href: '#', external: true, target: '_blank' } },
      ]}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();

  const link = await screen.findByRole<HTMLAnchorElement>('link');

  expect(link).toBeInTheDocument();
  expect(link.href).toBe('http://localhost/#');
  expect(link.target).toBe('_blank');
  expect(link.querySelector('svg')).toBeDefined();
});

test('renders header', () => {
  const { queryByText, container } = render(
    <Message header="Header" messages={[{ text: 'Success' }]} />,
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(getByText('Header')).toBeInTheDocument();
});

test('renders close button', () => {
  const { queryByRole, container } = render(
    <Message messages={[{ text: 'Success' }]} onClose={() => null} />,
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(getByRole('button')).toBeInTheDocument();
});

test('trigger onClose', async () => {
  const fn = jest.fn();

  render(<Message messages={[{ text: 'Success' }]} onClose={fn} />);

  const button = await screen.findByRole<HTMLButtonElement>('button');

  fireEvent.click(button);

  expect(fn).toHaveBeenCalled();
});

test('does not forward styles', () => {
  const { container } = render(
    <Message className="test" messages={[{ text: 'Success' }]} style={{ background: 'red' }} />,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('renders actions', () => {
  const onClick = jest.fn();
  const actions: MessageProps['actions'] = [
    { text: 'First Action', onClick },
    { text: 'Second Action', variant: 'secondary', onClick },
  ];

  const { container, getByRole } = render(
    <Message actions={actions} messages={[{ text: 'Success' }]} />,
  );
  const firstAction = getByRole('button', { name: 'First Action' });
  const secondAction = getByRole('button', { name: 'Second Action' });

  expect(container.firstChild).toMatchSnapshot();

  expect(firstAction).toHaveStyleRule('background-color', 'transparent');
  expect(secondAction).toHaveStyleRule('background-color', 'transparent');

  fireEvent.click(firstAction);

  expect(onClick).toHaveBeenCalledTimes(1);

  fireEvent.click(secondAction);

  expect(onClick).toHaveBeenCalledTimes(2);
});

test('renders localized labels', async () => {
  render(
    <Message
      localization={{ close: 'Cerrar' }}
      messages={[{ text: 'Success' }]}
      onClose={() => null}
    />,
  );

  const button = await screen.findByRole('button', { name: 'Cerrar' });

  expect(button).toBeInTheDocument();
});
