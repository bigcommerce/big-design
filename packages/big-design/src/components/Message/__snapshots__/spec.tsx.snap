// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render default (success) Message 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Success
      </span>
       
    </div>
  </div>
</div>
`;

exports[`render error Message 1`] = `
.c5 {
  vertical-align: middle;
  color: #DB3643;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #DB3643;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="error"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="danger"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1m1 4h-2v-2h2z"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Error
      </span>
       
    </div>
  </div>
</div>
`;

exports[`render info Message 1`] = `
.c5 {
  vertical-align: middle;
  color: #0B38D9;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #0B38D9;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="info"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="primary60"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 15c-.55 0-1-.45-1-1v-4c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1m1-8h-2V7h2z"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Info
      </span>
       
    </div>
  </div>
</div>
`;

exports[`render warning Message 1`] = `
.c5 {
  vertical-align: middle;
  color: #FFAE00;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #FFAE00;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="warning"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="warning50"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M4.47 21h15.06c1.54 0 2.5-1.67 1.73-3L13.73 4.99c-.77-1.33-2.69-1.33-3.46 0L2.74 18c-.77 1.33.19 3 1.73 3M12 14c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1m1 4h-2v-2h2z"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Warning
      </span>
       
    </div>
  </div>
</div>
`;

exports[`renders actions 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c9 {
  margin-top: 0.5rem;
  box-sizing: border-box;
}

.c10 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: background-color,border-color,box-shadow,color;
  transition-property: background-color,border-color,box-shadow,color;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
  background-color: transparent;
  border-color: #3C64F4;
  color: #3C64F4;
}

.c12.c12 {
  margin-bottom: 0.5rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.c12:focus {
  outline: none;
}

.c12[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c12 + .bd-button {
  margin-left: 0.5rem;
}

.c12:active {
  background-color: #DBE3FE;
}

.c12:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c12:hover:not(:active) {
  background-color: #F0F3FF;
}

.c12[disabled] {
  color: #D9DCE9;
}

.c13 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

.c11 {
  margin-bottom: -0.5rem;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

@media (min-width:720px) {
  .c12 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c12 {
    width: auto;
  }
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Success
      </span>
       
    </div>
    <div
      class="c9 c10 c11"
    >
      <button
        class="c12 bd-button"
      >
        <span
          class="c13"
        >
          First Action
        </span>
      </button>
      <button
        class="c12 bd-button"
      >
        <span
          class="c13"
        >
          Second Action
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders close button 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c12 {
  vertical-align: middle;
  height: 1.25rem;
  width: 1.25rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c9 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: background-color,border-color,box-shadow,color;
  transition-property: background-color,border-color,box-shadow,color;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  background-color: #3C64F4;
  border-color: #3C64F4;
  font-weight: 600;
}

.c9:focus {
  outline: none;
}

.c9[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c9 + .bd-button {
  margin-top: 0.5rem;
}

.c9:active {
  background-color: #0B38D9;
}

.c9:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c9:hover:not(:active) {
  background-color: #2852EB;
}

.c9[disabled] {
  background-color: #D9DCE9;
}

.c11 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
}

.c10 {
  background-color: transparent;
  border: none;
  color: #313440;
  height: auto;
  min-width: auto;
  padding: 0;
}

.c10:active {
  background-color: #DBE3FE;
}

.c10:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c10:hover:not(:active) {
  background-color: #F0F3FF;
}

.c10[disabled] {
  color: #D9DCE9;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages close';
  grid-template-columns: 1.5rem 1fr 1.25rem;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

@media (min-width:720px) {
  .c9 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c9 {
    width: auto;
    padding: 0;
    min-width: 2.25rem;
  }
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Success
      </span>
       
    </div>
  </div>
  <div
    class="c3 "
  >
    <button
      class="c9 c10"
    >
      <span
        class="c11"
      >
        <svg
          aria-labelledby=":ra:"
          class="c12"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <title
            id=":ra:"
          >
            Close
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`renders header 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
  margin: 0 0 0.5rem;
}

.c9 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c9:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  line-height: 1.25rem;
  margin-bottom: 0;
}

.c10 {
  color: #313440;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <h4
      class="c7 c8"
    >
      Header
    </h4>
    <div
      class="c3"
    >
      <span
        class="c9 c10"
      >
        Success
      </span>
       
    </div>
  </div>
</div>
`;

exports[`renders with external link 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c11 {
  vertical-align: middle;
  height: 1rem;
  width: 1rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c9 {
  -webkit-transition: all 70ms ease-out;
  transition: all 70ms ease-out;
  -webkit-transition-property: color;
  transition-property: color;
  color: #3C64F4;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9:active {
  color: #0024A6;
}

.c9:hover:not(:active) {
  color: #0024A6;
}

.c9 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 0.25rem;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

.c10 {
  font-size: 0.875rem;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Success
      </span>
       
      <a
        class="c9 c10"
        href="#"
        target="_blank"
      >
        <span>
          Link
        </span>
        <svg
          aria-hidden="true"
          class="c11"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M18 19H6c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5a2 2 0 0 0-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1s-1 .45-1 1v5c0 .55-.45 1-1 1M14 4c0 .55.45 1 1 1h2.59l-9.13 9.13a.996.996 0 1 0 1.41 1.41L19 6.41V9c0 .55.45 1 1 1s1-.45 1-1V4c0-.55-.45-1-1-1h-5c-.55 0-1 .45-1 1"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders with link 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
}

.c3 {
  box-sizing: border-box;
}

.c1 {
  gap: 1rem;
  display: grid;
}

.c4 {
  grid-area: icon;
}

.c6 {
  grid-area: messages;
}

.c9 {
  -webkit-transition: all 70ms ease-out;
  transition: all 70ms ease-out;
  -webkit-transition-property: color;
  transition-property: color;
  color: #3C64F4;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:active {
  color: #0024A6;
}

.c9:hover:not(:active) {
  color: #0024A6;
}

.c7 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c7:last-child {
  margin-bottom: 0;
}

.c2 {
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  grid-gap: 0.75rem;
  padding: 0.75rem;
  grid-template-areas: 'icon messages';
  grid-template-columns: 1.5rem 1fr;
  border-left: 0.25rem solid #2AAB3F;
}

.c8 {
  color: #313440;
  vertical-align: middle;
}

.c10 {
  font-size: 0.875rem;
  vertical-align: middle;
}

<div
  class="c0 c1 c2"
  role="alert"
  type="success"
>
  <div
    class="c3 c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
      />
    </svg>
  </div>
  <div
    class="c3 c6"
  >
    <div
      class="c3"
    >
      <span
        class="c7 c8"
      >
        Success
      </span>
       
      <a
        class="c9 c10"
        href="#"
      >
        Link
      </a>
    </div>
  </div>
</div>
`;
